app/src/main/java/com/arctouch/codechallenge/home/adapters/HomeAdapter.kt

```diff
@@ -0,0 +1,61 @@
+package com.arctouch.codechallenge.home.adapters
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.Navigation
+import androidx.paging.PagedListAdapter
+import androidx.recyclerview.widget.DiffUtil
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.home.fragments.MovieListFragment
+import com.arctouch.codechallenge.home.fragments.MovieListFragmentDirections
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.MovieImageUrlBuilder
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import kotlinx.android.synthetic.main.movie_item.view.*
+
+class HomeAdapter : PagedListAdapter<Movie, HomeAdapter.ViewHolder>(diffCallback) {
+
+    companion object {
+        private val diffCallback = object : DiffUtil.ItemCallback<Movie>() {
+            override fun areItemsTheSame(oldItem: Movie, newItem: Movie): Boolean =
+                    oldItem.id == newItem.id
+
+            override fun areContentsTheSame(oldItem: Movie, newItem: Movie): Boolean =
+                    oldItem == newItem
+
+        }
+    }
+
+    class ViewHolder(itemView: View) : androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView) {
+
+        private val movieImageUrlBuilder = MovieImageUrlBuilder()
+
+        fun bind(movie: Movie) {
+            itemView.titleTextView.text = movie.title
+            itemView.genresTextView.text = movie.genres?.joinToString(separator = ", ") { it.name }
+            itemView.releaseDateTextView.text = movie.releaseDate
```

Comment:
Do you think that formatting the releaseDate could bring a better user experience?

Reply:
Sure, it probably will!